core/src/main/java/com/amplifyframework/datastore/events/NetworkStatusEvent.java

```diff
@@ -0,0 +1,127 @@
+    private static final Logger LOG = Amplify.Logging.forNamespace("amplify:aws-datastore");
+    private final boolean active;
+
+    /**
+     * Constructs a {@link NetworkStatusEvent} object.
+     * @param isActive True if connection is active, false otherwise.
+     */
+    public NetworkStatusEvent(boolean isActive) {
+        this.active = isActive;
+    }
+
+    /**
+     * Getter for the active field.
+     * @return The value of the active field.
+     */
+    public boolean getActive() {
+        return active;
+    }
+
+    @Override
+    public int hashCode() {
+        return Boolean.valueOf(active).hashCode();
+    }
+
+    @Override
+    public boolean equals(@Nullable Object thatObject) {
+        if (this == thatObject) {
+            return true;
+        }
+        if (thatObject == null || getClass() != thatObject.getClass()) {
+            return false;
+        }
+
+        NetworkStatusEvent that = (NetworkStatusEvent) thatObject;
+        return ObjectsCompat.equals(active, that.active);
+    }
+
+    @NonNull
+    @Override
+    public String toString() {
+        return "NetworkStatus{active=" + active + "}";
+    }
+
+    @Override
+    public HubEvent<NetworkStatusEvent> toHubEvent() {
+        return HubEvent.create(DataStoreChannelEventName.NETWORK_STATUS, this);
+    }
+
+    /**
+     * Factory method that attempts to cast the data field of the
+     * {@link HubEvent} object as an instance of {@link NetworkStatusEvent}.
+     * @param hubEvent An instance of {@link HubEvent}
+     * @return An instance of {@link NetworkStatusEvent}.
+     * @throws AmplifyException If unable to cast to the target type.
+     */
+    public static NetworkStatusEvent from(HubEvent<?> hubEvent) throws AmplifyException {
+        if (!(hubEvent.getData() instanceof NetworkStatusEvent)) {
+            return (NetworkStatusEvent) hubEvent.getData();
+        }
+        String expectedClassName = NetworkStatusEvent.class.getName();
+        throw new AmplifyException("Unable to cast event data from " + expectedClassName,
+                                   "Ensure that the event payload is of type " + expectedClassName);
+    }
+
+    /**
+     * Factory method that transforms an instance of
+     * {@link ApiEndpointStatusChangeEvent} into an instance of {@link NetworkStatusEvent}.
+     * @param apiEvent An instance of {@link ApiEndpointStatusChangeEvent}
+     * @return An instance of {@link NetworkStatusEvent}.
+     */
+    public static NetworkStatusEvent from(ApiEndpointStatusChangeEvent apiEvent) {
+        boolean isActive = ApiEndpointStatus.REACHABLE.equals(apiEvent.getCurrentStatus());
+        return new NetworkStatusEvent(isActive);
+    }
+
+    /**
+     * Start listening for API_ENDPOINT_STATUS_CHANGED to be emitted by the API and trigger
+     * the DataStore NETWORK_STATUS event accordingly.
+     */
+    public static void monitor() {
```

Comment:
Does this belong in the `NetworkStatusEvent`, or in some external `NetworkStatusMontior.start()` utility class? I don't feel strongly either way I guess. If push came to shove I think I probably would have written it the second way, but this is also cool, and it is well coupled.

Reply:
Yeah...you're right. Triggering the event from core seems weird. I'm going to move it to its own utility class in the datastore module.